rules/ast/ast.go

```diff
@@ -241,12 +242,23 @@ type (
 	// least one of the two operand Nodes must be a VectorNode. The other may be
 	// a VectorNode or ScalarNode. Both criteria are checked at runtime.
 	VectorArithExpr struct {
-		opType BinOpType
-		lhs    Node
-		rhs    Node
+		opType        BinOpType
+		lhs           Node
+		rhs           Node
+		matchType     VectorMatchType
```

Comment:
I wonder if this would be better named `matchCardinality` (http://en.wikipedia.org/wiki/Cardinality_%28data_modeling%29) or `matchGrouping`. `Type` can mean so many things...